xbmc/video/VideoDatabase.cpp

```diff
@@ -3964,22 +3964,39 @@ void CVideoDatabase::GetSameVideoItems(CFileItem& item, CFileItemList& items)
 
     m_pDS->close();
 
-    // get items with same title
+    // get items with same title and year
     std::string title = item.GetVideoInfoTag()->GetTitle();
+    int year = item.GetVideoInfoTag()->GetYear();
+    bool hasYear = item.GetVideoInfoTag()->HasYear();
 
     VideoDbContentType itemType = item.GetVideoContentType();
     if (itemType == VideoDbContentType::MOVIES)
     {
+      std::vector<int> ids;
       m_pDS->query(PrepareSQL("SELECT idMovie FROM movie WHERE movie.c%02d = '%s'",
                               VIDEODB_ID_TITLE, title.c_str()));
 
       while (!m_pDS->eof())
       {
-        insertItemId(m_pDS->fv("idMovie").get_asInt());
+        ids.push_back(m_pDS->fv("idMovie").get_asInt());
         m_pDS->next();
       }
 
       m_pDS->close();
+
+      for (const auto id : ids)
+      {
+        CFileItem current;
+        if (GetDetailsByTypeAndId(current, itemType, id))
+        {
+          bool currentHasYear = current.GetVideoInfoTag()->HasYear();
+          int currentYear = current.GetVideoInfoTag()->GetYear();
+
+          // if both items have same year or don't have year
+          if ((hasYear && currentHasYear && year == currentYear) || (!hasYear && !currentHasYear))
```

Comment:
wouldn't just the condition `year == currentYear` basically cover all the other conditions?

Reply:
There are APIs for this, is it a good practice to use the implement details here?

Reply:
both methods have their use depending on the context they are needed. But here you only care if `year == year` or if `no year == no year` and those two cases are already covered by `year == year` because `getYear()` internally does the same checks as `hasYear()` and returns 0 if no year is present. So the complex condition could IMO be simplified, and if it is only to safe a few CPU cycles per video. That would at least be my take on it - not sure how others see this (I am by no means any code authority)

Reply:
This comparison (and orig PR code) compares the year and ignores the month/day part when it's available (premiered date with format YYYY-MM-DD instead of just year YYYY)

Should the compared items be required to both have premiered date or to both have only year? Or is it too strict?
I don't know much about scrapers, nfo or the db and it could be a stupid question ;-)
A comment would be nice if the month/day part is ignored on purpose.

For example is it the same item if the DB item has the premiered date 2023-11-28 and is compared with item that has only the year 2023?